src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -21,55 +44,748 @@ const ActorInit En_Insect_InitVars = {
+            }
+        }
+        currentActor = currentActor->next;
+    }
+    return ret;
+}
+
+void func_80A7C058(EnInsect* this) {
+    if (this->unk_31E > 0) {
+        this->unk_31E -= 1;
+        return;
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MUSI_WALK);
+
+    this->unk_31E = 3.0f / CLAMP_MIN(this->skelAnime.animPlaybackSpeed, 0.1f);
+    if (this->unk_31E < 2) {
+        this->unk_31E = 2;
+    }
+}
+
+void EnInsect_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnInsect* this = THIS;
+    f32 randomResult;
+    s16 temp_s2;
+    s32 count;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    func_80A7BE20(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BE6C.s")
+    temp_s2 = this->actor.params & 3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BF58.s")
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_04035590, &D_040341FC, this->limbDrawTable,
+                   this->transitionDrawTable, 24);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sColliderInit, &this->colliderItem);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BFA0.s")
+    this->actor.colChkInfo.mass = 30;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7C058.s")
+    if (this->unk_314 & 1) {
+        this->actor.gravity = -0.2f;
+        this->actor.minVelocityY = -2.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/EnInsect_Init.s")
+    if (this->unk_314 & 4) {
+        this->unk_31C = Math_Rand_S16Offset(200, 40);
+        this->actor.flags |= 0x10;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/EnInsect_Destroy.s")
+    if (temp_s2 == 2 || temp_s2 == 3) {
+        if (EnInsect_FindNearestSoil(this, globalCtx) != 0) {
+            this->unk_314 |= 0x10;
+            D_80A7DEB0 = 0.0f;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7C3A0.s")
+        if (temp_s2 == 2) {
+            this->actor.posRot.rot.z = 0;
+            this->actor.shape.rot.z = this->actor.posRot.rot.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7C3F4.s")
+            for (count = 0; count < 2; count++) {
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, this->actor.posRot.pos.x,
+                            this->actor.posRot.pos.y, this->actor.posRot.pos.z, this->actor.shape.rot.x,
+                            this->actor.shape.rot.y, this->actor.shape.rot.z, 3);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7C598.s")
+                if (globalCtx) {}
+            }
+        }
 
+        func_80A7D39C(this);
+
+        if (1) {}
+
+        D_80A7DEB8++;
+    } else {
+        randomResult = Math_Rand_ZeroOne();
```

Comment:
i tend to call these just "rand", its nice and short. its not a convention or anything, just a suggestion